lib/module.js

```diff
@@ -95,38 +96,83 @@ export default function SentryModule (moduleOptions) {
+  }
+
+  const apiMethods = await getBrowserApiMethods()
+
+  if (options.lazy) {
+    if (!options.lazy.injectMock) {
+      options.lazy.mockApiMethods = []
+    } else if (options.lazy.mockApiMethods === true) {
+      options.lazy.mockApiMethods = apiMethods
+    } else if (Array.isArray(options.lazy.mockApiMethods)) {
+      const mockMethods = options.lazy.mockApiMethods
+      options.lazy.mockApiMethods = mockMethods.filter(method => apiMethods.includes(method))
+
+      const notfoundMethods = mockMethods.filter(method => !apiMethods.includes(method))
+      if (notfoundMethods.length) {
+        logger.warn('Some specified methods to mock weren\'t found in @sentry/browser:', notfoundMethods)
+      }
+
+      if (!options.lazy.mockApiMethods.includes('captureException')) {
+        // always add captureException if a sentry mock is requested
+        options.lazy.mockApiMethods.push('captureException')
+      }
+    }
+  }
+
   const initializationRequired = options.initialize && options.dsn
 
   // Register the client plugin
-  if (!options.disabled && !options.disableClientSide) {
-    this.addPlugin({
-      src: resolve(__dirname, 'sentry.client.js'),
-      fileName: 'sentry.client.js',
-      mode: 'client',
-      options: {
-        config: {
-          dsn: options.dsn,
-          ...options.clientConfig
-        },
-        initialize: initializationRequired,
-        integrations: filterDisabledIntegration(options.clientIntegrations)
-          .reduce((res, key) => {
-            res[key] = options.clientIntegrations[key]
-            return res
-          }, {})
-      }
-    })
+  let pluginOptionClient
+  if (options.disabled || options.disableClientSide) {
+    pluginOptionClient = 'mocked'
+  } else if (options.lazy) {
+    pluginOptionClient = 'lazy'
   } else {
-    this.addPlugin({
-      src: resolve(__dirname, 'sentry.mocked.js'),
-      fileName: 'sentry.client.js',
-      options: { logMockCalls: options.logMockCalls },
-      mode: 'client'
-    })
+    pluginOptionClient = 'client'
   }
 
+  this.addPlugin({
+    src: resolve(__dirname, `plugin.${pluginOptionClient}.js`),
+    fileName: 'sentry.client.js',
+    mode: 'client',
+    options: {
+      dev: this.options.dev,
+      config: {
+        dsn: options.dsn,
+        ...options.clientConfig
+      },
+      lazy: options.lazy,
+      apiMethods,
+      logMockCalls: options.logMockCalls, // for mocked only
+      initialize: initializationRequired,
+      integrations: filterDisabledIntegration(options.clientIntegrations)
+        .reduce((res, key) => {
+          res[key] = options.clientIntegrations[key]
+          return res
+        }, {})
+    }
+  })
+
   // Register the server plugin
-  if (!options.disabled && !options.disableServerSide) {
+  let pluginOptionServer
+  if (options.disabled || options.disableServerSide) {
+    pluginOptionServer = 'mocked'
+  } else {
+    pluginOptionServer = 'server'
```

Comment:
With `disabled: true` (and `lazy` not set), we'll use "mocked" plugin for the server but we are not passing `apiMethods` and `logMockCalls` to it so things will crash.

(Can't mark the issue at proper line.)

(BTW. Enabling "mocked" plugin on the server is not really related to lazy loading but I guess it makes sense. Unless I'm missing something.)

Reply:
Have added the missing options.

Enabling the mocked plugin is whats happening now already right? I'm quite sure I only refactored this logic

Reply:
Yeah, you're right. It was like that before already. And it seems like the `logMockCalls` handling was kinda broken as we didn't pass that option before either.